File: license-tracker-backend/src/main/java/com/prodapt/license_tracker_backend/service/SoftwareVersionService.java
package com.prodapt.license_tracker_backend.service;


import com.prodapt.license_tracker_backend.dto.SoftwareVersionRequest;
import com.prodapt.license_tracker_backend.dto.SoftwareVersionResponse;
import com.prodapt.license_tracker_backend.dto.SoftwareVersionStatsResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface SoftwareVersionService {
    SoftwareVersionResponse createSoftwareVersion(SoftwareVersionRequest request);
    SoftwareVersionResponse updateSoftwareVersion(Long id, SoftwareVersionRequest request);
    void deleteSoftwareVersion(Long id);
    SoftwareVersionResponse getSoftwareVersionById(Long id);
    Page<SoftwareVersionResponse> getAllSoftwareVersions(Pageable pageable);
    List<SoftwareVersionResponse> getSoftwareVersionsByDevice(Long deviceId);
    List<SoftwareVersionResponse> getSoftwareVersionsByStatus(String status);
    SoftwareVersionStatsResponse getStatistics();
    SoftwareVersionResponse checkForUpdates(Long id);
}

File: licensetracker_frontend/src/services/softwareVersionService.js
import api from './api';

const softwareVersionService = {
    getAllSoftwareVersions: async (page = 0, size = 10, sortBy = 'lastChecked') => {
        const response = await api.get(`/software-versions?page=${page}&size=${size}&sortBy=${sortBy}`);
        return response.data;
    },

    getSoftwareVersionById: async (id) => {
        const response = await api.get(`/software-versions/${id}`);
        return response.data;
    },

    getSoftwareVersionsByDevice: async (deviceId) => {
        const response = await api.get(`/software-versions/by-device/${deviceId}`);
        return response.data;
    },

    getSoftwareVersionsByStatus: async (status) => {
        const response = await api.get(`/software-versions/by-status/${status}`);
        return response.data;
    },

    getStatistics: async () => {
        const response = await api.get('/software-versions/statistics');
        return response.data;
    },

    createSoftwareVersion: async (data) => {
        const response = await api.post('/software-versions', data);
        return response.data;
    },

    updateSoftwareVersion: async (id, data) => {
        const response = await api.put(`/software-versions/${id}`, data);
        return response.data;
    },

    checkForUpdates: async (id) => {
        const response = await api.post(`/software-versions/${id}/check-updates`);
        return response.data;
    },

    deleteSoftwareVersion: async (id) => {
        const response = await api.delete(`/software-versions/${id}`);
        return response.data;
    }
};

export default softwareVersionService;
